dcpy/builds/load.py

```diff
@@ -58,17 +102,65 @@ def load_source_data(
 
     build_name = metadata.build_name()
     logger.info(f"Loading source data for {recipe.name} build named {build_name}")
-
-    pg_client = postgres.PostgresClient(schema=build_name)
-    setup_build_environments(pg_client)
-
-    pg_client.create_table_from_csv(
-        recipe_lock_path.parent / "source_data_versions.csv"
+    if plan.InputDatasetDestination.postgres.value in [
+        dataset.destination for dataset in recipe.inputs.datasets
+    ]:
+        pg_client = postgres.PostgresClient(schema=build_name)
+        setup_build_environments(pg_client)
```

Comment:
It looks like `setup_build_environments` also deletes the s3 draft folder for the product. Do we need to do that with other destinations? (In any case, I'm more in favor of just breaking that function up into its s3 and pg component parts)

Reply:
Ooh that's a good catch. Yes, I'll break up the logic in there

Reply:
Do we actually want to be deleting s3 artifacts here? Functionality in `edm-publishing` deletes folder when loading to s3. If my build errors out in the middle of a build but prior to loading, I'd prefer the previous build's output don't vanish. If you agree, I'm just going to scrap that bit of code

Reply:
Good question... It does seem really misplaced. Scrapping it seems good to me